src/main/java/org/truffleruby/cext/CExtNodes.java

```diff
@@ -1055,17 +1059,66 @@ public DynamicObject classNew(VirtualFrame frame, DynamicObject superclass) {
     @CoreMethod(names = "rb_tr_debug", onSingleton = true, rest = true)
     public abstract static class DebugNode extends CoreMethodArrayArgumentsNode {
 
-        @TruffleBoundary
+        @Child CallDispatchHeadNode toSCall;
+
         @Specialization
-        public Object debug(Object... objects) {
+        public Object debug(VirtualFrame frame, Object... objects) {
             if (objects.length > 1) {
-                System.err.printf("Printing %d values%n", objects.length);
+                printf("Printing %d values%n", objects);
             }
             for (Object object : objects) {
-                System.err.printf("%s @ %s: %s%n", object.getClass(), System.identityHashCode(object), object);
+                final String representation;
+                if (RubyGuards.isRubyString(object)) {
+                    representation = getRubyStringRepresentation((DynamicObject) object);
+                } else if (RubyGuards.isRubyBasicObject(object)) {
+                    representation = getString(object) + " (" +
+                            RopeOperations.decodeRope(StringOperations.rope(callToS(frame, object))) + ")";
```

Comment:
You could use `StringOperations.getString` here.

Reply:
nice, done